core/src/test/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationEndpointFilterTest.java

```diff
@@ -0,0 +1,380 @@
+import static org.mockito.Mockito.when;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.http.HttpStatus;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.keygen.StringKeyGenerator;
+import org.springframework.security.oauth2.core.OAuth2ErrorCodes;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest;
+import org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames;
+import org.springframework.security.oauth2.server.authorization.OAuth2Authorization;
+import org.springframework.security.oauth2.server.authorization.OAuth2AuthorizationService;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClient;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClientRepository;
+import org.springframework.security.oauth2.server.authorization.client.TestRegisteredClients;
+import org.springframework.security.oauth2.server.authorization.util.OAuth2AuthorizationServerMessages;
+import org.springframework.security.web.RedirectStrategy;
+
+
+/**
+ * Tests for {@link OAuth2AuthorizationEndpointFilter}.
+ *
+ * @author Paurav Munshi
+ * @since 0.0.1
+ */
+
+public class OAuth2AuthorizationEndpointFilterTest {
+
+	private static final String VALID_CLIENT = "valid_client";
+	private static final String VALID_CLIENT_MULTI_URI = "valid_client_multi_uri";
+	private static final String VALID_CC_CLIENT = "valid_cc_client";
+
+	private OAuth2AuthorizationEndpointFilter filter;
+
+	private RedirectStrategy authorizationRedirectStrategy = mock(RedirectStrategy.class);
+	private Converter<HttpServletRequest, OAuth2AuthorizationRequest> authorizationConverter = mock(Converter.class);
+	private OAuth2AuthorizationService authorizationService = mock(OAuth2AuthorizationService.class);
+	private StringKeyGenerator codeGenerator = mock(StringKeyGenerator.class);
+	private RegisteredClientRepository registeredClientRepository = mock(RegisteredClientRepository.class);
+	private Authentication authentication = mock(Authentication.class);
+
+	@Before
+	public void setUp() {
+		filter = new OAuth2AuthorizationEndpointFilter();
+
+		filter.setAuthorizationService(authorizationService);
+		filter.setCodeGenerator(codeGenerator);
+		filter.setRegisteredClientRepository(registeredClientRepository);
+
+		SecurityContextHolder.getContext().setAuthentication(authentication);
+	}
+
+	@Test
+	public void testFilterRedirectsWithCodeOnValidReq() throws Exception {

```

Comment:
There is a naming convention for test methods. For an example, please see `DefaultOAuth2AuthorizationRequestResolverTests` and `OAuth2AuthorizationCodeGrantFilterTests`

Reply:
Fixed in PR. Thanks.